packages/vscode-extension/src/webview/components/SendFilesOverlay.tsx

```diff
@@ -0,0 +1,158 @@
+import { useState, useMemo, useEffect, useCallback } from "react";
+import { VscodeProgressRing } from "@vscode-elements/react-elements";
+import "./Preview.css";
+import "./SendFilesOverlay.css";
+import { use$ } from "@legendapp/state/react";
+import { useProject } from "../providers/ProjectProvider";
+import classNames from "classnames";
+import { useSelectedDeviceSessionState } from "../hooks/selectedSession";
+
+const RETAIN_SUCCESS_SCREEN = 1000; // ms
+const RETAIN_ERROR_SCREEN = 3000; // ms
+
+// Important! You need to hold shift to drag files onto the panel
+// VSCode displays a "Hold shift to drop into editor" message when Preview is in the Editor Tab
+// but it doesn't show this when Preview is in the Side Panel
+export function SendFilesOverlay() {
+  const { project } = useProject();
+  const [isVisible, setIsVisible] = useState(false);
```

Comment:
I think it'd be better to have a "isDragging" state that you set in the drag events, and derive "isVisible" from it and the other possible states:

```suggestion
  const [isDragging, setIsDragging] = useState(false);
  const isVisible = isDragging || isSuccess || isError;
```